### DISCUSSION TITLE
Hyphen don't work anymore

### DISCUSSION THREAD

**Discussion by guigro** [YOURLS/YOURLS#3328] [Plugins]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/YOURLS/.github/blob/main/CODE_OF_CONDUCT.md).

### Submission validity

- [X] This is not a personal support request, that should be posted on [discussions community](https://github.com/YOURLS/YOURLS/discussions).
- [X] I checked [current issues](https://github.com/YOURLS/YOURLS/issues) and this request isn't a duplicate of an existing issue, opened or closed.

### Self troubleshooting

- [X] I've read the [documentation](https://yourls.org/) and made sure to follow it.
- [X] I've read the [troubleshooting first steps](https://docs.yourls.org/guide/troubleshooting/first-steps.html) and [frequent issues](https://docs.yourls.org/guide/troubleshooting/frequent-issues.html) guides.

### Version

1.9

### Description

For a few days, my shortener wasn't working.
It's all based on the acceptation of hyphens in my shortened link.

I updated, and it's still not working. 
If I don't activate the plugin "allow hyphens", it deletes the hyphens in my links, if I activate the plugin, i get a "YOURLS has nothing for you to see here."

I tried to deleted my database and reinstall the software, no change.

### Expectation

I was expecting my URL to be redirected correctly

### Reproduction steps

1. Install YOURLS
2. Activate Allow Hyphen plugin
3. Try to shorten an URL

**Comment by dgw** [comment]
This plugin hasn't changed since YOURLS 1.6, and the filter it uses hasn't changed since 1.8.

You have indicated that you read the troubleshooting steps, but have not provided any debug output or other logs to help diagnose the problem.

**Comment by NetJon** [comment]
@guigro 

Did you ever figure this out?

I just upgraded to 1.9 and am having problems with the Allow Hyphens plugin.

Strangely, I cannot even find that plugin here on GitHub anywhere.

Were you able to get your Yourls installation working again?

@dgw - I see you listed as the "maintainer."  Where is Allow Hyphens even at?  I would like to poke around and see if my problems are already addressed before opening an issue, although I cannot even find the plugin to do that.

**Reply by dgw** [reply]
I am _a_ maintainer, but not _the_ only one. :)

The plugin to allow hyphens in short URLs is part of core: https://github.com/YOURLS/YOURLS/tree/master/user/plugins/hyphens-in-urls

**Reply by NetJon** [reply]
Thank you @dgw

**Reply by guigro** [reply]
Hello @NetJon !
I didn't managed yet to solve this issue.
I tried clean install, as it was working before and stopped working I guess when I upgraded.
I'll try again and come back here with result !

**Reply by NetJon** [reply]
I got rid of a couple of plugins that had not been updated for years and that seemed to fix the problem.  Maybe there is a conflict?

One was called ssl4ssl and the other was an outdated version of a plugin to block the recording of clicks from a logged in user in the stats.

With those gone, it appears to work fine now for me on two different installations.

**Reply by guigro** [reply]
Yep I agree ! I made a fresh install and activated just that plugin. I can't explain how it's working now as whatever I did before it was not working.

But I'm happy now it's all fine !!!! :)

**Reply by dgw** [reply]
@strarsis Please do not comment on random old issues about this. I'm mentioning you here in reply to those off-topic comments.

Allow Hyphens is part of core. It comes with YOURLS in the install archive.

**Reply by strarsis** [reply]
@dgw: Thanks! Though when I enter a short URL with hyphens, the hyphens are stripped. Latest YOURLS `v1.9.1`.

**Reply by dgw** [reply]
@strarsis Did you activate the plugin?

**Reply by strarsis** [reply]
@dgw: No, actually not. So this plugin needs to be activated by placing it into the plugin folder or by the admin GUI?

**Reply by dgw** [reply]
@strarsis GUI

**Reply by strarsis** [reply]
@dgw: Found it! Though that only active plugins were listed below the manage plugins link confused me as I thought the plugin listed below is the only available one (but actually the only active one, more are available that aren't active)...

**Comment by YildirimErdemli** [comment]
Hi people, I started using yourls today and I also experienced the hypen problem. Here's what needs to be done. You must activate the Allow Hyphens in Short URLs plugin. Then, if you want to use it as **my awesome special short url**, you should write it as **my-awesome-special-short-url**. So it records with hypens.

**Reply by NetJon** [reply]
I think this thread is from almost 2 years ago.

**Reply by YildirimErdemli** [reply]
Yes that's right however the problem is new to me because I encountered it yesterday. Since there was no answer, I tried to solve it and wrote it in case it might help someone. I hope this is not wrong. Thanks.

**Reply by guigro** [reply]
Even two years later : thanks a lot !!

**Reply by NetJon** [reply]
@YildirimErdemli 

Not wrong that I know of.

I just didn't know if you had noticed the age of the thread.

But as far as I know, there are no rules against commenting on old threads.  😎

**Reply by YildirimErdemli** [reply]
Yes, I am aware, thank you. It's still an active project and as I said, I just installed and used it. I just wanted to have the answer while I found the solution. I'd be happy if someone who needs it finds the answer.
